benchmark/single-source/KeyPathPerformanceTests.swift

```diff
@@ -0,0 +1,1349 @@
+    self.element89 = element89
+    self.element90 = element90
+    self.element91 = element91
+    self.element92 = element92
+    self.element93 = element93
+    self.element94 = element94
+    self.element95 = element95
+    self.element96 = element96
+    self.element97 = element97
+    self.element98 = element98
+    self.element99 = element99
+  }
+
+  var element0: Element
+  var element1: Element
+  var element2: Element
+  var element3: Element
+  var element4: Element
+  var element5: Element
+  var element6: Element
+  var element7: Element
+  var element8: Element
+  var element9: Element
+  var element10: Element
+  var element11: Element
+  var element12: Element
+  var element13: Element
+  var element14: Element
+  var element15: Element
+  var element16: Element
+  var element17: Element
+  var element18: Element
+  var element19: Element
+  var element20: Element
+  var element21: Element
+  var element22: Element
+  var element23: Element
+  var element24: Element
+  var element25: Element
+  var element26: Element
+  var element27: Element
+  var element28: Element
+  var element29: Element
+  var element30: Element
+  var element31: Element
+  var element32: Element
+  var element33: Element
+  var element34: Element
+  var element35: Element
+  var element36: Element
+  var element37: Element
+  var element38: Element
+  var element39: Element
+  var element40: Element
+  var element41: Element
+  var element42: Element
+  var element43: Element
+  var element44: Element
+  var element45: Element
+  var element46: Element
+  var element47: Element
+  var element48: Element
+  var element49: Element
+  var element50: Element
+  var element51: Element
+  var element52: Element
+  var element53: Element
+  var element54: Element
+  var element55: Element
+  var element56: Element
+  var element57: Element
+  var element58: Element
+  var element59: Element
+  var element60: Element
+  var element61: Element
+  var element62: Element
+  var element63: Element
+  var element64: Element
+  var element65: Element
+  var element66: Element
+  var element67: Element
+  var element68: Element
+  var element69: Element
+  var element70: Element
+  var element71: Element
+  var element72: Element
+  var element73: Element
+  var element74: Element
+  var element75: Element
+  var element76: Element
+  var element77: Element
+  var element78: Element
+  var element79: Element
+  var element80: Element
+  var element81: Element
+  var element82: Element
+  var element83: Element
+  var element84: Element
+  var element85: Element
+  var element86: Element
+  var element87: Element
+  var element88: Element
+  var element89: Element
+  var element90: Element
+  var element91: Element
+  var element92: Element
+  var element93: Element
+  var element94: Element
+  var element95: Element
+  var element96: Element
+  var element97: Element
+  var element98: Element
+  var element99: Element
+
+  static func getKeypathToElement(index: Int)
```

Comment:
I would put an `@inline(never)` on this function to rule out that it's getting inlined, which would mean that the keypath will be completely optimized away.

Reply:
Interesting. Yes, adding the `@inline(never)` did actually get the run times to be the about the same between "9" and "44" for `lastKeyPathIndex`. (Release mode now takes over a minute to compile!) It's looking more like this is a non-issue and if so, I can pull these benchmarks out in the next commit.

Reply:
Please don't forget to add the `@inline(never)` here

Reply:
Ok, that should be in now.